Assets/BossRoom/Scripts/Client/UI/Lobby/JoinLobbyUI.cs

```diff
@@ -0,0 +1,172 @@
+                m_UpdateRunner.Unsubscribe(PeriodicRefresh);
+            }
+        }
+
+        [Inject]
+        private void InjectDependencies(IInstanceResolver container, LobbyUIMediator lobbyUIMediator, LobbyServiceData lobbyServiceData, UpdateRunner updateRunner)
+        {
+            m_Container = container;
+            m_LobbyUIMediator = lobbyUIMediator;
+            m_LobbyServiceData = lobbyServiceData;
+            m_UpdateRunner = updateRunner;
+
+            m_UpdateRunner.Subscribe(PeriodicRefresh, 10f);
+            BeginObserving(m_LobbyServiceData);
+        }
+
+        public void OnJoinButtonPressed()
+        {
+            m_LobbyUIMediator.JoinLobbyWithCodeRequest(m_JoinCodeField.text.ToUpper());
+        }
+
+        private void PeriodicRefresh(float _)
+        {
+            m_LobbyUIMediator.QueryLobbiesRequest(false);
+        }
+
+        public void OnRefresh()
+        {
+            m_LobbyUIMediator.QueryLobbiesRequest(true);
+        }
+
+        protected override void UpdateObserver(LobbyServiceData observed)
+        {
+            base.UpdateObserver(observed);
+
+            ///Check for new entries, We take CurrentLobbies as the source of truth
+            var previousKeys = new List<string>(m_LobbyButtons.Keys);
+
+            foreach (var codeLobby in observed.CurrentLobbies)
+            {
+                var lobbyCodeKey = codeLobby.Key;
+                var lobbyData = codeLobby.Value;
+
+                if (!m_LobbyButtons.ContainsKey(lobbyCodeKey))
+                {
+                    if (CanDisplay(lobbyData))
+                    {
+                        CreateLobbyPanel(lobbyCodeKey, lobbyData);
+                    }
+                }
+                else
+                {
+                    if (CanDisplay(lobbyData))
+                    {
+                        UpdateLobbyButton(lobbyCodeKey, lobbyData);
+                    }
+                    else
+                    {
+                        RemoveLobbyButton(lobbyData);
+                    }
+                }
+
+                previousKeys.Remove(lobbyCodeKey);
+            }
+
+            foreach (var key in previousKeys)
+                // Need to remove any lobbies from the list that no longer exist.
+            {
+                RemoveLobbyButton(m_LocalLobby[key]);
+            }
+        }
+
+        public void OnQuickJoinClicked()
+        {
+            m_LobbyUIMediator.QuickJoinRequest();
+        }
+
+        private bool CanDisplay(LocalLobby lobby)
+        {
+            return !lobby.Private && lobby.LobbyUsers.Count != lobby.MaxPlayerCount;
+        }
+
+        /// <summary>
+        ///     Instantiates UI element and initializes the observer with the LobbyData
+        /// </summary>
+        private void CreateLobbyPanel(string lobbyCode, LocalLobby lobby)
+        {
+            var lobbyPanel = Instantiate(m_LobbyPanelPrototype.gameObject, m_LobbyPanelPrototype.transform.parent)
```

Comment:
Pretty sure you could just do Instantiate(m_LobbyPanelPrototype, m_LobbyPanelPrototype.transform.parent);
And not have to do a get component and a .gameObject